packages/keyring/src/pair/index.ts

```diff
@@ -176,8 +176,9 @@ export function createPair ({ toSS58, type }: Setup, { publicKey, secretKey }: P
     toJson: (passphrase?: string): KeyringPair$Json => {
       // NOTE: For ecdsa and ethereum, the publicKey cannot be extracted from the address. For these
       // pass the hex-encoded publicKey through to the address portion of the JSON (before decoding)
+      // unless the publicKey is already an address
       const address = ['ecdsa', 'ethereum'].includes(type)
-        ? u8aToHex(secp256k1Compress(publicKey))
+        ? publicKey.length === 21 ? u8aToHex(secp256k1Compress(publicKey)) : u8aToHex(publicKey)
```

Comment:
I am not sure in which cases `=== 21` would be true here? The publicKey would be longer than that and the address is exact 20.

Reply:
Are you sure it's not 21 with the prefix? This worked when copying on the apps.
Just like 33 is for 32 bytes

Reply:
So the raw Uint8Array is 20 bytes for Ethereum addresses. 

The ecdsa public is indeed 33 (normal Substrate is 32)

Reply:
Ok makes sens